src/commands/batch_read_command.rs

```diff
@@ -145,12 +145,6 @@ impl<'a, 'b> BatchReadCommand<'a, 'b> {
                         .get_mut(batch_record.batch_index)
                         .expect("Invalid batch index");
                     let record = batch_record.record;
-                    if let Some(Record {
-                        key: Some(ref key), ..
-                    }) = record
-                    {
-                        assert_eq!(batch_read.key.digest, key.digest);
-                    }
                     batch_read.record = record;
```

Comment:
Seems like the temp. `record` variable is no longer needed, and this could be simplified to

    batch_read.record = batch_record.record

Reply:
good catch, let me fix that

Reply:
good catch, let me fix that